File: MORM.Java/src/MORM/Java/Classes/ProviderFactory.java
package MORM.Java.Classes;

import java.sql.Connection;
import java.sql.DriverManager;

public class ProviderFactory
{
	public static String GetConnectionStringByProvider(String providerName)
	{
		String connectionString = null;
		for (TipoDatabase tipoDatabase : TipoDatabase.values())
			if (tipoDatabase.GetProviderName().equals(providerName))
				connectionString = tipoDatabase.GetConnectionString();
		return connectionString;
	}
	
	public static Connection CreateConnection(
		String providerName, String connectionString, String username, String password)
	{
		Connection connection = null;
		try {
			Class.forName(providerName);
			connection = DriverManager.getConnection(connectionString, username, password);
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
		} catch (Exception e) {
			e.printStackTrace();
		}		
		return connection;
	}
}

File: MORM.CSharp/src/MORM.Repositorio/Factories/ProviderFactory.cs
using System;
using System.Configuration;
using System.Data.Common;

namespace MORM.Repositorio
{
    public static class ProviderFactory
    {
        public static string GetConnectionString(string providerName)
        {
            // Return null on failure.
            string returnValue = null;
        
            // Get the collection of connection strings.
            ConnectionStringSettingsCollection settings = ConfigurationManager.ConnectionStrings;
        
            // Walk through the collection and return the first 
            // connection string matching the providerName.
            if (settings != null)
            {
                foreach (ConnectionStringSettings cs in settings)
                {
                    if (cs.ProviderName == providerName)
                    {
                        returnValue = cs.ConnectionString;
                        break;
                    }
                }
            }
            
            return returnValue;
        }
        
        public static DbConnection GetConnection(string providerName, string connectionString)
        {
            // Assume failure.
            DbConnection connection = null;
        
            // Create the DbProviderFactory and DbConnection.
            if (connectionString != null)
            {
                try
                {
                    DbProviderFactory factory = DbProviderFactories.GetFactory(providerName);
        
                    connection = factory.CreateConnection();
                    connection.ConnectionString = connectionString;
                }
                catch (Exception ex)
                {
                    // Set the connection to null if it was created.
                    if (connection != null)
                        connection = null;
                    Console.WriteLine(ex.Message);
                }
            }

            // Return the connection.
            return connection;
        }
    }
}